src/plugin/barcode-scan.js

```diff
@@ -95,13 +96,20 @@ function processKeyEvent(e) {
         previousKey.receivedAt = 0;
     }
 
-    e.preventDefault();
-    e.stopImmediatePropagation();
-    e.stopPropagation();
+    // prevent native keydown behavior if we detected space
+    // or we are in time based scan (at least two chars are added to final result)
+    if (inScanMode || scanResult.code.length >= 2) {
+        logger.out('debug', '[GGP] Plugin Barcode: Prevent keydown default behavior.');
+        e.preventDefault();
+        e.stopImmediatePropagation();
+        e.stopPropagation();
+    }
 
     // we need moove focus out of any input to body so we don't enter 
-    // codes from scaner but we don't know if first char is from scan so don't move focus when first code is entered
-    if (scanResult.code.length !== 0 && document.activeElement && document.activeElement.tagName.toLocaleLowerCase() !== 'body') {
+    // codes from scaner (e.g we could trigger enter on payin)
+    // but we don't know if first char is from scan so don't move focus when first code is entered
+    if ((scanResult.code.length >= 2 || inScanMode) &&
```

Comment:
Since these ifs have repeating condition maybe we can move this in some helper function 

`const isScanModeActive = () => (inScanMode || scanResult.code.length >= 2);`